lib/devicetrust/native/device_windows.go

```diff
@@ -0,0 +1,489 @@
+// Copyright 2023 Gravitational, Inc
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package native
+
+import (
+	"crypto/rsa"
+	"crypto/sha256"
+	"crypto/x509"
+	"encoding/base64"
+	"math/big"
+	"os"
+	"os/exec"
+	"os/user"
+	"path"
+	"strings"
+
+	"github.com/google/go-attestation/attest"
+	"github.com/gravitational/trace"
+	log "github.com/sirupsen/logrus"
+	"google.golang.org/protobuf/types/known/timestamppb"
+
+	devicepb "github.com/gravitational/teleport/api/gen/proto/go/teleport/devicetrust/v1"
+	"github.com/gravitational/teleport/lib/devicetrust"
+)
+
+const (
+	deviceStateFolderName  = ".teleport-device"
+	attestationKeyFileName = "attestation.key"
+)
+
+// Ensures that device state directory exists.
```

Comment:
nit: even though it's not public:

```suggestion
// setupDeviceStateDir ensures that device state directory exists.
```